edgePy/_DGEList.py

```diff
@@ -120,6 +120,10 @@ def samples(self, samples: Optional[np.ndarray]) -> None:
         # - Samples same length as ncol(self.counts) if defined
         if samples is not None:
             samples = np.array(list(self._format_fields(samples)))
+
+            if self.counts is not None:
+                if self.counts.shape[1] != len(samples):
```

Comment:
Why not:

```python
if self.counts is not None and len(samples) != self.counts.shape[1]:
    raise ...
```

This keeps the if statements un-nested. This second equality expression also reads better with your statements reversed.

Reply:
Indeed! I'll fix that.